include/BatteryGuard.h

```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+#pragma once
+#include <Arduino.h>
+#include <frozen/string.h>
+#include <TaskSchedulerDeclarations.h>
+#include "Statistic.h"
+
+
+class BatteryGuardClass {
+    public:
+        BatteryGuardClass() = default;
+        ~BatteryGuardClass() = default;
+        BatteryGuardClass(const BatteryGuardClass&) = delete;
+        BatteryGuardClass& operator=(const BatteryGuardClass&) = delete;
+        BatteryGuardClass(BatteryGuardClass&&) = delete;
+        BatteryGuardClass& operator=(BatteryGuardClass&&) = delete;
+
+        void init(Scheduler& scheduler);
+        void updateSettings(void);
+        void updateBatteryValues(float const nowVoltage, float const nowCurrent, uint32_t const millisCurrent);
+        bool isInternalResistanceCalculated(void) const;
+        std::optional<float> getOpenCircuitVoltage(void);
+        std::optional<float> getInternalResistance(void) const;
+
+    private:
+        void loop(void);
+        void slowLoop(void);
+
+        Task _slowLoopTask;                                 // Task (print the report)
+        Task _fastLoopTask;                                 // Task (get the battery values)
+        bool _verboseLogging = false;                       // Logging On/Off
+        bool _verboseReport = false;                        // Report On/Off
```

Comment:
@SW-Niko Is it fine for you if we merge those two config flags into a single `_verboseLogging`? Or is there a strong reason to have those separated?

```suggestion
        bool _verboseLogging = false;                       // Logging/Report On/Off
```

Reply:
I think it is no longer a problem. I had a problem in the past with to many logging's. 
But is seems that issue is fixed

Reply:
Danke :)